src/Module.ts

```diff
@@ -439,9 +466,25 @@ export default class Module {
 
 		// we don't want to create shims when we are just
 		// probing export * modules for exports
-		if (this.graph.shimMissingExports && !isExportAllSearch) {
-			this.shimMissingExport(name);
-			return this.exportShimVariable;
+		if (!isExportAllSearch) {
+			if (this.syntheticNamedExports) {
+				let syntheticExport = this.syntheticExports.get(name);
+				if (!syntheticExport && !this.exports[name]) {
+					const defaultExport = this.getDefaultExport();
+					if (defaultExport) {
```

Comment:
This check should be superfluous as `this.getDefaultExport` throws when it cannot find an export. Therefore, the check `if (syntheticExport)` below is also unneeded and this can be simplified. Furthermore, the test for `!this.exports[name]` should always be true as otherwise, we would not get here.